cdisc_rules_engine/models/rule_conditions/condition_composite.py

```diff
@@ -24,13 +25,53 @@ class ConditionComposite(ConditionInterface):
     def __init__(self):
         self._conditions = {}
 
+    def get_conditions(self) -> dict:
+        return self._conditions
+
+    def set_target(self, target) -> "ConditionInterface":
+        for key, conditions_list in self._conditions.items():
+            new_conditions_list = []
+            for condition in conditions_list:
+                new_conditions_list.append(condition.copy().set_target(target))
+            self._conditions[key] = new_conditions_list
+        return self
+
+    def set_conditions(self, conditions: dict):
+        self._conditions = conditions
+
+    def copy(self):
+        new_condition = ConditionComposite()
+        for key, condition in self._conditions.items():
+            new_condition.add_conditions(key, condition)
+        return new_condition
+
     def add_conditions(self, key: str, conditions: List[ConditionInterface]):
         """
         Adds a list of objects with ConditionInterface
         interface to the given key.
         """
         self._conditions[key] = conditions
 
+    def add_conditions_for_targets(self, targets: List[str]):
+        """
+        If a condition specifies the parameter variable: "all",
+        the condition will be duplicated for all targets.
+        """
+        if self.should_copy():
+            for key, conditions in self._conditions.items():
+                conditions_to_add = []
+                for cond in conditions:
+                    if isinstance(cond, SingleCondition):
```

Comment:
Please get rid of if/else here and try to use Composite pattern idea

Reply:
Ahh I meant to remove that function altogether. It’s no longer used